go/types/struct.go

```diff
@@ -180,31 +181,38 @@ var fieldNameRe = regexp.MustCompile(fieldNameComponentRe.String() + "$")
 type encodingFunc func(string, *regexp.Regexp) string
 
 func CamelCaseFieldName(input string) string {
+	//strip invalid struct characters and leave spaces
 	encode := func(s1 string, p *regexp.Regexp) string {
-		if p.MatchString(s1) {
+		if p.MatchString(s1) || spaceRegex.MatchString(s1) {
 			return s1
 		}
 		return ""
 	}
 
-	splitFields := strings.Fields(input)
-	output := escapeFields(splitFields[0], encode)
+	strippedField := escapeField(input, encode)
+	splitField := strings.Fields(strippedField)
+	output := ""
 
-	if len(splitFields) > 1 {
-		output = strings.ToLower(output)
+	//Camelcase field
+	if len(splitField) == 1 {
+		output = splitField[0]
```

Comment:
discussed irl, lowercasing this makes sense especially because is a camelcasing function